packages/frontend/workspace/src/providers/awareness/broadcast-channel/index.ts

```diff
@@ -0,0 +1,63 @@
+import type { Awareness } from 'y-protocols/awareness.js';
+import {
+  applyAwarenessUpdate,
+  encodeAwarenessUpdate,
+} from 'y-protocols/awareness.js';
+
+import type { AwarenessProvider } from '..';
+import type { AwarenessChanges } from '../affine';
+
+type ChannelMessage =
+  | { type: 'connect' }
+  | { type: 'update'; update: Uint8Array };
+
+export function createBroadcastChannelAwarenessProvider(
```

Comment:
will this make doc receives duplicated awareness update from both cloud and other tabs in same channel?

Reply:
no, because both broadcast and cloud mark received updates as `origin:'remote'`